public/app/plugins/datasource/grafana-azure-monitor-datasource/components/MetricsQueryEditor/MetricNamespaceField.tsx

```diff
@@ -38,6 +38,12 @@ const MetricNamespaceField: React.FC<MetricNamespaceFieldProps> = ({
   if (value && !optionValues.includes(value.toLowerCase())) {
     options.push({ label: value, value });
   }
+  const noAdditionalNamespaces =
+    metricNamespaces.length === 1 && metricNamespaces[0].value === query.azureMonitor?.metricNamespace;
+
+  if (noAdditionalNamespaces) {
+    return null;
```

Comment:
I know I suggested this change but now I am not that sure :thinking: My reasoning is:

 - We are not sure we are covering all the cases, we may have another bug in the API not listing the right namespaces and this would prevent people from manually setting the namespace.
 - Even if there is only one namespace, people may want to use a template variable for it.
 - Azure portal still shows the selector if there is only one namespace, it's probably a good idea to keep the same experience:

![Screenshot from 2022-09-13 10-12-59](https://user-images.githubusercontent.com/4025665/189850616-154a69ee-0db7-4637-a5e6-78921b4051a6.png)

Thoughts?

Reply:
I would still do it if you only had points number 1 and 3 , 

I thinking complicating the UX for the sake of keeping a workaround in case we ever have a bug or for consistency with Azure is not reason enough.

However, you make a good point about template variables, I do not use them enough to know about this use case. I understand how you would use a template variable for the resource group / subscription, possibly metrics, but I fail to find a example where a template variable would be used for metric namespace (that's very possibly because I've only ever seen one namespace per resource).

All in all, even if I like the approach that simplify the UI slightly better, I don't feel super strongly either way, the dropdown is automatically filled with the namespace so, if there is only one, it won't add a click for the user.

Happy for it to go either way 👍

Reply:
yep, the other two points are more subjective but removing the possibility of using template variables could be a breaking change